test/end-to-end/offer.js

```diff
@@ -1417,6 +1454,125 @@ describe("Offer endpoint tests", () => {
+                                    expect(res.body.results[i].publishDate)
+                                        .toBe(res.body.results[i + 1].publishDate);
+
+                                    // eslint-disable-next-line no-undef
+                                    expect(BigInt(`0x${res.body.results[i]._id}`))
+                                        // eslint-disable-next-line no-undef
+                                        .toBeLessThan(BigInt(`0x${res.body.results[i + 1]._id}`));
+                                }
+                            }
+                        });
+
+                        test("Should return next most recent offer", async () => {
+                            const res1 = await test_agent
+                                .get("/offers")
+                                .query({
+                                    limit: 3
+                                });
+
+                            expect(res1.status).toBe(HTTPStatus.OK);
+                            expect(res1.body.results).toHaveLength(3);
+
+                            const res2 = await test_agent
+                                .get("/offers")
+                                .query({
+                                    queryToken: res1.body.queryToken,
+                                    limit: 1
+                                });
+
+                            expect(res2.status).toBe(HTTPStatus.OK);
+                            expect(res2.body.results).toHaveLength(1);
+                            expect((new Date(res2.body.results[0].publishDate)).getTime())
+                                .toBeLessThan((new Date(res1.body.results[2].publishDate)).getTime());
+                        });
+
+                        test("Should return next offer that is as recent but with a higher id", async () => {
+                            const res1 = await test_agent
+                                .get("/offers")
+                                .query({
+                                    limit: 2
+                                });
+
+                            expect(res1.status).toBe(HTTPStatus.OK);
+                            expect(res1.body.results).toHaveLength(2);
+
+                            const res2 = await test_agent
+                                .get("/offers")
+                                .query({
+                                    queryToken: res1.body.queryToken,
+                                    limit: 1
+                                });
+
+                            expect(res2.status).toBe(HTTPStatus.OK);
+                            expect(res2.body.results).toHaveLength(1);
+                            // eslint-disable-next-line no-undef
+                            expect(BigInt(`0x${res2.body.results[0]._id}`))
+                                // eslint-disable-next-line no-undef
+                                .toBeGreaterThan(BigInt(`0x${res1.body.results[1]._id}`));
```

Comment:
I'm a bit confused by these comparisons, why do they need to be converted into BigInts?

Reply:
Before converting to BigInts I tried converting to Numbers. When doing so, there was a loss of precision when doing the conversion. The numerical value of the id capped at approximately 3.0907e+28 (between 2^94 and 2^95), and ids can go up to 2^96-1, since each id is a 12-byte value. Hence the BigInt conversion to have the necessary precision.
However, I've put some more thought into the matter and realized that I could be comparing the strings directly, since digits come before letters in lexicographical order and each id is guaranteed to be 12 bytes long. The snippet would look like this:

```suggestion
                            expect(`0x${res.body.results[i]._id}` < `0x${res.body.results[i + 1]._id}`)
                                            .toBeTruthy();
```

Reply:
Should i submit these changes or keep it as it is?

Reply:
Doesn't the "greaterThan" function (I don't remember the exact name) work with strings?

Reply:
No, it only works with numeric values (Number and BigInt)

Reply:
Does the change you suggested work?

Reply:
Yes

Reply:
I like that snippet a lot more. Also, you can remove 0x and get rid of string interpolation.

Reply:
The snippet's version is implemented in all tests that used the BigInt conversion